src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,33 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture
+                .supplyAsync(
+                        () -> cache.computeIfAbsent(userId, this::getUserData),
+                        executor
+                );
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.SECONDS.sleep(1);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
Catching InterruptedException is necessary when a thread is interrupted during a blocking operation. However, it is important to handle this exception properly. Simply throwing a new RuntimeException can disrupt the thread's interrupted status. Consider re-interrupting the thread with Thread.currentThread().interrupt() to preserve the interrupt status after catching the exception.